src/wagtail_vector_index/storage/django.py

```diff
@@ -371,8 +445,21 @@ def bulk_generate_documents(self, objects, *, embedding_backend):
                     content=chunk,
                 )
 
-        # Return every document object, regardless of whether it was rebuilt, retaining
-        # the order they appeared in the original list
+    async def _acreate_new_documents_bulk(
+        self, *, documents_by_object, objects_by_key, all_chunks
+    ):
+        for object_key, documents in documents_by_object.items():
+            for idx, returned_embedding in documents:
+                all_keys = self._keys_for_instance(objects_by_key[object_key])
+                chunk = all_chunks[idx]
+                await Document.objects.acreate(
```

Comment:
Do we want to be creating them one at a time rather than than in bulk? This feels like it'll do a lot of moving work onto a sync thread per `acreate` call when we could just use `bulk_create` once